app/controllers/doorkeeper/authorizations_controller.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require_dependency "#{Doorkeeper::Engine.root}/app/controllers/doorkeeper/authorizations_controller.rb"
+
+module Doorkeeper
+  class AuthorizationsController
+    module AuthorizationsExtension
+      private
+
+      def pre_auth_param_fields
+        super.append(:nonce)
```

Comment:
@linhdangduy this might be because the Doorkeeper controllers can be customized: https://github.com/doorkeeper-gem/doorkeeper/wiki/Customizing-routes#changing-controllers

IIRC this was also the reason I monkey-patched `Doorkeeper::Helpers::Controller` instead in https://github.com/doorkeeper-gem/doorkeeper-openid_connect/blob/master/lib/doorkeeper/openid_connect/helpers/controller.rb.

It's not very nice either, but WDYT about moving `pre_auth_param_fields` into that file as well? This method is only used by `authorizations_controller.rb` so we don't need to perform any checks for the current controller, as in `authenticate_resource_owner!`.

Reply:
> @linhdangduy this might be because the Doorkeeper controllers can be customized: https://github.com/doorkeeper-gem/doorkeeper/wiki/Customizing-routes#changing-controllers
> 
> IIRC this was also the reason I monkey-patched `Doorkeeper::Helpers::Controller` instead in https://github.com/doorkeeper-gem/doorkeeper-openid_connect/blob/master/lib/doorkeeper/openid_connect/helpers/controller.rb.

User can customized Doorkeeper controller, but (I think) at least they will inherit their controller by doorkeeper's controller. So maybe monkey-patching the `AuthorizationsController` ok !?

> It's not very nice either, but WDYT about moving `pre_auth_param_fields` into that file as well? 

This is helpers module but the method is used only at one controller.
But anw, I can move that method to `Doorkeeper::Helpers::Controller`.

> This method is only used by `authorizations_controller.rb` so we don't need to perform any checks for the current controller, as in `authenticate_resource_owner!`.

Because `nonce` is introduced in openid-connect document, so to be sufficiently, we can perform check whether the scope param has `openid` or not, if it has, add `nonce` to permitted params. Should we? 🤔

Reply:
Okay let's keep this as it is, I don't really see a way to extend this more cleanly! :+1: And yes I think it's fine to always permit `nonce`.